src/main/java/org/folio/rest/impl/InventoryHelper.java

```diff
@@ -626,4 +643,61 @@ private JsonObject getFirstObjectFromResponse(JsonObject response, String proper
                    .map(item -> (JsonObject) item)
                    .orElseThrow(() -> new CompletionException(new InventoryException(String.format("No records of '%s' can be found", propertyName))));
   }
+
+  /**
+   * Returns JsonObject with id of ContributorNameType retrieved from inventory-storage.
+   *
+   * @param contributorNameTypeName ContributorNameType
+   * @return JsonObject ContributorNameTypeId
+   */
+  private CompletableFuture<JsonObject> getContributorNameTypeId(ContributorNameTypeName contributorNameTypeName) {
+    return cacheAndGet(contributorNameTypeName.getName(),
```

Comment:
As you are using function as handler you might pass function to construct endpoint whe needed like
```java
  private CompletableFuture<JsonObject> getContributorNameTypeId(ContributorNameTypeName contributorNameTypeName) {
    UnaryOperator<String> endpoint = query -> String.format("/contributor-name-types?query=name==%s", encodeQuery(query, logger));
    return cacheAndGet(contributorNameTypeName.getName(), endpoint,
      entries -> {
        JsonObject contributorPersonalNameType = new JsonObject();
        contributorPersonalNameType.put(CONTRIBUTOR_NAME_TYPE_ID, getFirstObjectFromResponse(entries, CONTRIBUTOR_NAME_TYPES).getString(ID));
        return contributorPersonalNameType;
      });
  }
...
  private CompletableFuture<JsonObject> cacheAndGet(String key, UnaryOperator<String> endpoint, Function<JsonObject, JsonObject> handler) {
    JsonObject response = ctx.get(key);
    if(response == null) {
      return handleGetRequest(endpoint.apply(key), httpClient, ctx, okapiHeaders, logger).thenApply(json -> {
        JsonObject result = handler.apply(json);
        ctx.put(key, result);
        return result;
      });
    } else {
      CompletableFuture<JsonObject> future = new CompletableFuture<>();
      future.complete(ctx.get(key));
      return future;
    }
  }
```